src/1a_prep_MPA.jl

```diff
@@ -1,189 +1,298 @@
 """
-Prepare data for analysis.
+Prepare data for analysis by processing MPA files for each GBRMPA management region.
 
 Crop GBR-wide GBRMPA rasters into management regions.
-Reproject all data from WGS84 / UTM Zone 54 - 56 into consistent crs GDA-2020.
-Ensure all rasters are the same size/shape for each region of interest.
+Reproject all data from WGS84 / UTM Zone 54 - 56 into consistent CRS (GDA2020).
+Ensure all rasters are the same size/shape for each region of interest with the same
+values used to indicate no data.
+
+The general approach is to crop down to an extent of the region, then trim/mask to just the
+areas of interest. Processing data in this way minimized the amount of data being handled.
+
+Still, the largest chunk of time is spent resampling to the same projections and writing
+data to file (as it takes time to compress the data). The datasets can be in the GBs without
+compression.
 """
 
 include("common.jl")
 
-# 1. Processing of GBRMPA zoning geopackage to select only zones for site exclusion
+# Loading regions_4326 for cropping of vector and raster data.
+regions_4326 = GDF.read(REGION_PATH_4326)
+
+# 1. Process GBRMPA zoning geopackage to select only zones for site exclusion
 if !isfile(joinpath(MPA_OUTPUT_DIR, "GBRMPA_zone_exclusion.gpkg"))
-    GBRMPA_zoning_poly = GDF.read(joinpath(GDA2020_DATA_DIR, "Great_Barrier_Reef_Marine_Park_Zoning_20_4418126048110066699.gpkg"))
+    GBRMPA_zoning_poly = GDF.read(
+        joinpath(
+            GDA2020_DATA_DIR,
+            "Great_Barrier_Reef_Marine_Park_Zoning_20_4418126048110066699.gpkg"
+        )
+    )
     GBRMPA_zoning_poly = GBRMPA_zoning_poly[GBRMPA_zoning_poly.TYPE .∈ [MPA_EXCLUSION_ZONES], :]
     rename!(GBRMPA_zoning_poly, :SHAPE => :geometry)
 
-    GDF.write(joinpath(MPA_OUTPUT_DIR, "GBRMPA_zone_exclusion.gpkg"), GBRMPA_zoning_poly; crs=EPSG(7844))
+    GDF.write(
+        joinpath(MPA_OUTPUT_DIR, "GBRMPA_zone_exclusion.gpkg"),
+        GBRMPA_zoning_poly;
+        crs=EPSG(7844)
+    )
 end
 
-# 2. Processing of MPA files into smaller GDA-2020 rasters
-# Loading regions_4326 for cropping of vector and raster data.
-regions_4326 = GDF.read(REGION_PATH_4326)
+# 2. Process MPA files to represent GBRMPA regions in GDA2020 projection
+
+"""
+    set_consistent_missingval(raster, val)
```

Comment:
Just curious about why this vs just `= replace_missing!()`

Reply:
`replace_missing!()` replaces the value used to indicate `no data`, but does not update the metadata.

e.g., if the no data value was -1, `replace_missing!(raster, -2)` will change all `-1` it finds to `-2`

`Raster()` with a defined `missingval` loads the dataset with the indicated value set as the `no data` value, but does not change the data.

e.g., if the no data value is -1, calling `Raster(filepath; missingval=-2)` will load the data as usual, but it will not change the `-1`s in the dataset to `-2`.

This helper function replaces the value and recreates the raster so the metadata matches the given missing value.

Reply:
Ok that makes sense to ensure the resulting .tif file has the right no data value.